src/audio/google_rtc_audio_processing.c

```diff
@@ -371,50 +372,59 @@ static int google_rtc_audio_processing_copy(struct comp_dev *dev)
 
 	buffer_stream_invalidate(cd->aec_reference, num_aec_reference_bytes);
 
-	aec_reference_buff_frag = 0;
-	for (frame = 0; frame < num_aec_reference_frames; frame++) {
-		for (channel = 0; channel < cd->num_aec_reference_channels; ++channel) {
-			src = audio_stream_read_frag_s16(&cd->aec_reference->stream,
-							 aec_reference_buff_frag + channel);
-			cd->aec_reference_buffer[cd->num_aec_reference_channels *
-				cd->aec_reference_frame_index + channel] = *src;
+	num_samples_remaining = num_aec_reference_frames * cd->num_aec_reference_channels;
```

Comment:
Note that the number of AEC reference channels passed to the AEC might be lower than the number of channel in the stream. See comment in prepare about that.
Should be like:
`num_samples_remaining = num_aec_reference_frames * cd->aec_reference->stream.channels;`